compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/SyntheticAccessorLowering.kt

```diff
@@ -49,6 +55,53 @@ private class SyntheticAccessorLowering(val context: JvmBackendContext) : IrElem
 
     override fun lower(irFile: IrFile) {
         inlineLambdaToCallSite.putAll(InlineReferenceLocator.scan(context, irFile).lambdaToCallSite)
+
+        // Unconditionally add bridges for hidden constructors
+        irFile.transformChildrenVoid(object: IrElementTransformerVoid() {
```

Comment:
Why not override `visitConstructor` & `visitConstructorCall` in `SyntheticAccessorLowering` itself, instead of an extra object?

Reply:
Because we need to generate bridges unconditionally, even if there is no call to the constructor in the current module.